### PR TITLE
Add basic redis setup for rate limiting

### PR DESCRIPTION
**Pull request by GraemeHarris**
As per https://github.com/LAION-AI/Open-Assistant/issues/175 redis setup to be used for rate limiting - maybe BE cache down the line. 

I have updated the docker compose file to include redis + redis insights. They're quite basic for now with a lot of settings that can be tweaked as per need / feedback. 

Ansible changes might need more testing

### CODE REVIEW

File: docker-compose.yaml [hunk 1]
```diff
@@ -27,6 +27,29 @@ services:
...
+    volumes:
+      - ./redis.conf:/usr/local/etc/redis/redis.conf
+      - redis:/data
+  # insights host - redis:6379
+  redis-insights:
+    image: redislabs/redisinsight:latest
+    ports:
+      - 8001:8001
+    volumes:
+      - redis:/data
```
Comment (yk) [inline docker-compose.yaml:51]: do you think it's necessary to define a volume here? seems like for a test setup it's fine if the data goes down with the container, especially bc we don't use this redis as long-term storage

**Comment by yk** [comment]
also, some merge conflicts

File: docker-compose.yaml (hunk 1 above)
Comment (GraemeHarris) [reply docker-compose.yaml]: Fair point. Removed 👍

**Review by yk — CHANGES_REQUESTED** [review]
could you remove the ansible part? so we can get this merged. can you also make a new issue for including redis in ansible? I'll take that myself and do it later.

**Comment by andreaskoepf** [comment]
@yk is this ready for merge?

**Comment by kiritowu** [comment]
@GraemeHarris do you intend to add `redis` (and `redis-insights`) to Line 7 in the `docker-compose.yml` such that both `redis` and `redis-insights` will be started up for `backend-dev` purpose?

**Review by yk — APPROVED** [review]
Thank you!

**Pull request merged**